### PR TITLE
Minor integration tests adjustments

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

ADP-493

# Overview

- 1b26c2b9f965e8012d14882c107934ca6442787d
  Stronger assertion for ADDRESS_LIST_05 - bech32 HRP is correct - mainnet
  
- 5c526a6a3efbce38fa5db505d5250c784736b329
  Extra balance check for WALLETS_COIN_SELECTION_05

- 1b26c2b9f965e8012d14882c107934ca6442787d
  Stronger assertion for ADDRESS_LIST_05 - bech32 HRP is correct - mainnet
  
- 5c526a6a3efbce38fa5db505d5250c784736b329
  Extra balance check for WALLETS_COIN_SELECTION_05
  
- 5495755607f13d31fa57ad3e477dfc419deeb7a8
  TRANS_EXTERNAL_01/02 - remove potential flakiness
  
- 6538063a6d5ac1c3fed70d4fb2f02c58dbce699c
  Make CoinSelections separate test suites not to mess with Jormungandr integration tests

# Comments

### CODE REVIEW

**Review by paweljakubas — APPROVED** [review]

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Addresses.hs [hunk 1]
```diff
@@ -236,6 +236,8 @@ spec = describe "SHELLEY_ADDRESSES" $ do
             -- integration tests are configured for mainnet
             , expectListField 0 (Aeson.key "id" . Aeson._String)
                 (`shouldSatisfy` T.isPrefixOf "addr")
+            , expectListField 0 (Aeson.key "id" . Aeson._String)
+                (`shouldNotSatisfy` T.isPrefixOf "addr_test")
```
Comment (rvl) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Addresses.hs:240]: Oops - thanks for fixing.

**Review by rvl — APPROVED** [review]
Thanks

**Comment by piotr-iohk** [comment]
bors try

**Review by paweljakubas — APPROVED** [review]
looks like sound code reshuffling plus reasonable code additions

**Comment by piotr-iohk** [comment]
bors try

**Pull request merged**